pkg/componentsregistry/componentsregistry.go

```diff
@@ -0,0 +1,49 @@
+// componentsregistry package is a registry of all components that can be managed by the operator
+// TODO: it may make sense to put it under components/ when it's clear from the old stuff
+package componentsregistry
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-multierror"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dscv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/datasciencecluster/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+)
+
+// ComponentHandler is an interface to manage a component
+// Every method should accept ctx since it contains the logger.
```

Comment:
@ykaliuta Is this comment accurate? It looks like only one of the function signatures here in the interface accepts ctx

Reply:
> @ykaliuta Is this comment accurate? It looks like only one of the function signatures here in the interface accepts ctx

It's sort of TODO/good to have. Current code did not do it and with the introduction of the infra I left decision about it for later.

Reply:
Ah right, thanks :+1: